### PR TITLE
[CHE-193] Create and Seed Mongo Container for Dev

### PR DESCRIPTION
**Pull request by seantokuzo**
### Description

As part of the General Improvements story this PR aims to enhance the Codehammers™ developer experience. What it do?
- Define a mongo database service in `docker-compose-dev` - this service creates an instance of MongoDB running in a docker container usable by our application in development
- Define all necessary environmental variables for the dev container in its docker-compose file - no more need for a `.env` file
- Creates scripts that are placed into the mongo dev container that seed the dev database with mock data on startup
- Creates a named volume for mongo data. This allows data to persist if you stop and restart the mongo-dev service. This data persistence enables the mongo-dev container to skip the seeding phase, making subsequent startups quicker
- Exposes port 3000 on the development container for manual testing with Postman

### Jira Task

[JIRA TICKET](https://codehammers.atlassian.net/browse/CHE-193?atlOrigin=eyJpIjoiNDEwYTIwODYzZmY1NDQ2ZWEzNWNiOTUxN2UxMjg3YTgiLCJwIjoiaiJ9)

### Testing Instructions

- run `npm run docker-remove-all` to remove existing containers/images/volumes
- run `npm run docker-build-check` to check linting and unit tests still pass
- run `npm run docker-dev`
- navigate to [http://localhost:8080](http://localhost:8080)
- login with test user: email = `tester@codehammers.com`, password = `ilovetesting`
- navigate to alumni/profiles/forums pages and check that mock data has populated

__For fun__
- While docker-dev is running open up a terminal
- Run `docker exec -it ch-mongo-dev mongosh -u admin -p adminpassword` - this will allow you to run commands with `mongosh` inside of the docker container
- run `use ch-testdb` to switch to our development database
- run `db.users.find()` - this should return all of our mock users that have been seeded into the development database
- run the same command as above but with any other collection name (eg. `db.<collection_name>.find()`) - our collection names are: `users`, `alumnis`, `graduateinvitations`, `profiles`, `forums`, `threads`, `posts`
- you can also query for specific documents: eg. `db.users.find({email: "tester@codehammers.com"})`

#### All Team Members

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [ ] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [x] I ran `npm run docker-test` in my local environment to check that this PR passes all unit tests.
- [x] I did a quick check to make sure my code changes follow the recomended style guide.

### CODE REVIEW

**Review by serenahromano — APPROVED** [review]
nice!

File: docker-compose-dev.yml [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
 services:
   dev:
-    #image: codehammers/ch-dev:latest
-    #image: codehammers/ch-dev-dep-v2:latest
-    #image: codehammers/ch-tracker-dev:latest
     image: codehammers/ch-dev-dep-v3:latest
     container_name: ch-dev
     ports:
       - '8080:8080'
+      - '3000:3000'
```
Comment (brok3turtl3) [inline docker-compose-dev.yml:8]: 👍

File: scripts/db/mongo-dev-init/MOCK_USERS.json [hunk 2]
```diff
@@ -0,0 +1,1719 @@
...
+      "$oid": "666cbc4240af7b375e352e8c"
+    },
+    "lastVisit": {
+      "$date": "2024-06-14T21:55:14.814Z"
+    },
+    "date": {
+      "$date": "2024-06-14T21:55:14.814Z"
+    },
+    "__v": 0
+  },
```
Comment (brok3turtl3) [inline scripts/db/mongo-dev-init/MOCK_USERS.json]: Hey @seantokuzo  - Nice to see the site has so many active users! While Testy McTesterson seems like a huge asset to the team, can we add a few teammates for them? Specifically, let's add Jane@codehammers.com, john@codehammers.com, and jaime@ codehammers.com. They're all great friends so don't mind sharing the same "ilovetesting" password.

**Review by brok3turtl3 — CHANGES_REQUESTED** [review]
Love the scripts here @seantokuzo - nice work! 🎉 

❔ We can continue the conversation about this outside this ticket but a 37k line file leaves me uneasy lol...

🛑 Need to add our three core testing accounts listed in the readme to our seeder data and attribute some of the threads and posts to them

🛑 When I spin things up as per the instructions none of the image pulls work. Attaching some ss's. Is there a way to get your stock images to show up in dev mode when the S3 credentials have been disabled?
Screenshot 2024-06-16 at 8 38 30 AM
Screenshot 2024-06-16 at 8 48 09 AM

**Review by jimpulsee — APPROVED** [review]

**Review by brok3turtl3 — CHANGES_REQUESTED** [review]
Hey @seantokuzo - Solid workaround to display stock images in dev mode! The user is still able to attempt an upload of a new image in EditProfile. This ends up breaking the display and we also get some S3 connection errors in the terminal as expected with fake credentials. 
Screenshot 2024-06-19 at 6 44 29 AM
Screenshot 2024-06-19 at 6 51 06 AM

**Pull request merged**